app/api/search/route.ts

```diff
@@ -0,0 +1,19 @@
+import { NextResponse } from "next/server";
+import { useSearchParams } from "next/navigation";
+import discogs from "@/app/utils/discogs";
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url);
+  console.log(request.url);
+  const query = searchParams.get("search");
```

Comment:
you can just use the [nextUrl object here](https://nextjs.org/docs/app/api-reference/functions/next-request#nexturl):

```suggestion
  const query = request.nextUrl.searchParams.search;
```